lib/jekyll/drops/drop.rb

```diff
@@ -137,7 +137,16 @@ def inspect
         require 'json'
         JSON.pretty_generate to_h
       end
-      alias_method :to_json, :inspect
+
+      # When a JSON representation of the drop is requested (e.g. through
+      # template filters) then only show the YAML front matter data.
+      # Other Jekyll objects might not be dumpable due to recursiveness.
+      #
+      # Returns a JSON dump of the YAML front matter data.
+      def to_json
+        require 'json'
+        JSON.generate fallback_data
```

Comment:
* Prefer to use `to_json` instead of `JSON.generate` unless you intend to use `JSON.pretty_generate`